octodns/__init__.py

```diff
@@ -1,6 +1,25 @@
-'OctoDNS: DNS as code - Tools for managing DNS across multiple providers'
-
 from __future__ import absolute_import, division, print_function, \
     unicode_literals
+import pkg_resources
+from os import path
```

Comment:
It's personal preference, but I like to do `from os.path import dirname, join` so that the invocations later are much less verbose. Seems like doing things that way here would prevent a lot of the line wrapping that's necessary.